apisix/plugins/api-breaker.lua

```diff
@@ -35,6 +35,13 @@ local schema = {
             minimum = 200,
             maximum = 599,
         },
+        break_response_body = {
+            type = "string"
+        },
+        break_response_content_type = {
```

Comment:
Apache APISIX has a few plugins that define the response body but without setting the content type.

I'm thinking about that, if we need to keep the behaviors consistence?

Reply:
If the body is json, and the content-type is text/html, some APP clients may report error, and the terminal user would be effected.

Reply:
This problem cannot be solved even if we provide this field. People have to know how to configure it.

For these kinds of returning body, I think we may need a unified way to handle the `Content-Type`.

Reply:
> This problem cannot be solved even if we provide this field. People have to know how to configure it.
> 
> For these kinds of returning body, I think we may need a unified way to handle the `Content-Type`.

A unified way is good, but it's not that easy. Modern API returns application/json, but some legacy API returns application/soap+xml like SOAP.

Reply:
I think the API owner should be responsible to configure route on the apigageway correctly , it's part of the developing work, and only the API owner knows what the correct Content-Type should be. If there's no way to config the Content-Type correctly, when break state is open, the client APP used this API may crash because of the unexpected Content-Type.

Reply:
Fair enough. But would you like to submit an issue to track the content-type field for other plugins?

Reply:
> Fair enough. But would you like to submit an issue to track the content-type field for other plugins?

Ok, i'll browse all the plugins which should be with a content-type field and submit an issue with a checklist.

Reply:
> Ok, i'll browse all the plugins which should be with a content-type field and submit an issue with a checklist.

It is not a good idea, as this check won't fix the non-open-source plugins.

One solution is to guess:
1. according to the response body: https://github.com/apache/apisix/blob/cc88a8db0ba64e1cb351e389ca5ec64c516073d6/apisix/control/router.lua#L81 or using https://github.com/spacewander/lua-resty-mime-sniff
2. according to the Accept header or the ext part of the uri

Another solution is to configure it in the route

Reply:
As for this plugin, I would suggest adding a field to add any response headers instead of just content-type.

Reply:
> As for this plugin, I would suggest adding a field to add any response headers instead of just content-type.

Is there any other plugin witch such a field? I'm not sure of which field is suitable for representing any response headers.

Reply:
> As for this plugin, I would suggest adding a field to add any response headers instead of just content-type.

Can we use the ```headers``` field to represent any response headers? I found the ```echo``` plugin and also the ```response-rewrite``` plugin using this attribute. @spacewander 

Name | Type | Requirement | Default | Valid | Description
-- | -- | -- | -- | -- | --
headers | object | optional |   |   | New headers for response

Reply:
Yes. And we should use an array so multiple response headers with the same name can be used.